samples/sync_query.py

```diff
@@ -0,0 +1,33 @@
+from __future__ import print_function  # For python 2/3 interoperability
+from samples.utils import get_service, query_paging, poll_query
+
+
+# [START sync_query]
+def sync_query(service, project_id, query, timeout=0):
+    """ Run a synchronous query"""
+    query_data = {
+                    'query': query,
+                    'timeoutMs': timeout,
+                 }
+
+    return service.jobs().query(
+            projectId=project_id,
+            body=query_data).execute()
```

Comment:
google api client lib provides num_retries keyword argument, which will retry the request upon intermittent failure.

This is extremely useful especially for idempotent operation and maybe you can advertise it.